include/boost/random/faure.hpp

```diff
@@ -0,0 +1,369 @@
+/* boost random/faure.hpp header file
+ *
+ * Copyright Justinas Vygintas Daugmaudis 2010-2017
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_RANDOM_FAURE_HPP
+#define BOOST_RANDOM_FAURE_HPP
+
+#include <boost/random/detail/qrng_base.hpp>
+
+#include <vector>
+#include <algorithm>
+
+#include <cmath>
+
+#include <boost/assert.hpp>
+
+#include <sstream>
+
+//!\file
+//!Describes the quasi-random number generator class template faure.
+
+namespace boost {
+namespace random {
+
+/** @cond */
+namespace detail {
+namespace fr {
+
+// There is no particular reason why 187 first primes were chosen
+// to be put into this table. The only reason was, perhaps, that
+// the number of dimensions for Faure generator would be around
+// the same number as the number of dimensions supported by the
+// Sobol qrng.
```

Comment:
I'd rather see as many dimensions supported as possible, letting each qrng have it's own reasonable maximum size.